llvm/tools/llvm-cas-object-format/llvm-cas-object-format.cpp

```diff
@@ -634,27 +661,44 @@ void StatCollector::printToOuts(ArrayRef<CASID> TopLevels) {
   llvm::sort(Kinds);
 
   size_t NumHashBytes = TopLevels.front().getHash().size();
-  outs() << "  => Note: 'Parents' counts incoming edges\n"
-         << "  => Note: 'Children' counts outgoing edges (to sub-objects)\n"
-         << "  => Note: HashSize = " << NumHashBytes << "B\n"
-         << "  => Note: PtrSize  = " << sizeof(void *) << "B\n"
-         << "  => Note: Cost     = Count*HashSize + PtrSize*Children + Data\n";
-  StringLiteral HeaderFormat = "{0,-22} {1,+10} {2,+7} {3,+10} {4,+7} {5,+10} "
-                               "{6,+7} {7,+10} {8,+7} {9,+10} {10,+7}\n";
-  StringLiteral Format = "{0,-22} {1,+10} {2,+7:P} {3,+10} {4,+7:P} {5,+10} "
-                         "{6,+7:P} {7,+10} {8,+7:P} {9,+10} {10,+7:P}\n";
-  outs() << llvm::formatv(HeaderFormat.begin(), "Kind", "Count", "", "Parents",
-                          "", "Children", "", "Data (B)", "", "Cost (B)", "");
-  outs() << llvm::formatv(HeaderFormat.begin(), "====", "=====", "",
-                          "=======", "", "========", "", "========", "",
-                          "========", "");
-  auto printInfo = [Format, NumHashBytes, &Totals = this->Totals](
-                       StringRef Kind, ObjectKindInfo Info) {
+
+  raw_fd_ostream &outStream = printToStdout ? outs() : outCSV();
```

Comment:
Good idea, let me see what that looks like

Reply:
Probably the parameter can be a `raw_ostream` since none of the users care that it's related to file descriptor.

Reply:
I didn't make this change because main calls to `computeStats`, which then calls `printToOuts` So I would have to thread the argument through two function calls, with it having no use in `computeStats`

But that is just my preference, please let me know if you disagree, and I will make the change

Reply:
That doesn't seem too bad. Alternatively, you could refactor to make `computeStats` to return a result, and then call a second function to print them.

Reply:
I went with the former!